### ISSUE TITLE
Java 10 segfaults on latest generation AWS instances

### ISSUE THREAD

**Issue by xose** [elastic/elasticsearch-docker#179]
* Docker image used: `docker.elastic.co/elasticsearch/elasticsearch:6.3.0`

## Bug Description

Running the 6.3.0 Docker image on the latest-generation AWS instances (C5, M5, etc) crashes Java with a segfault when indexing data. See elastic/elasticsearch#31425 for details.

I have tracked it down to the version of Java that ships with the 6.3.0 Docker image. Downgrading Java to the JRE provided by the base image (java-1.8.0-openjdk-headless) fixes the issue.

While this is a Java/Lucene bug, it might be advisable to downgrade the JRE version on the official Docker image or release both -jre8 and -jre10 variants until the underlying issue is resolved.

## Workaround

Manually rebuild the Docker image with Java 8.

**Comment by dliappis** [comment]
Some preliminary investigation going on in https://markmail.org/message/n7rjurxzinyrhrwb / cc @jpountz 

@xose Do you happen to have a workflow/load pattern that we can use to reproduce this on the instances you mentioned?

**Comment by xose** [comment]
The new machines were nearly empty when the crashes started happening. I updated the `index.routing.allocation.require.*` for a small active index (around 650 docs indexed/sec) and `.monitoring-*` to move their shards to the new c5d machines for testing, and ~10 minutes later got the first crash followed by more crashes every few minutes on different machines. I then moved the indexes back to the original i3 instances.

Problematic instances: 3x c5d.2xlarge running Amazon Linux ECS 2018.03.a
Disk: 200GB NVMe instance store, formatted as XFS and mounted on ES data directory
ES Settings: 6GB heap, 30% index_buffer_size (Other system settings as recommended on [Production Mode](https://www.elastic.co/guide/en/elasticsearch/reference/current/docker.html#docker-cli-run-prod-mode))

Index settings on the active index: translog.durability=async, refresh_interval=10s
